File: test/cpp/case25_for_range.cpp
#include "cpp_module/gc.h"
#include "cpp_module/py_runtime_modules.h"
#include <algorithm>
#include <any>
#include <fstream>
#include <ios>
#include <iostream>
#include <sstream>
#include <stdexcept>
#include <string>
#include <tuple>
#include <type_traits>
#include <unordered_map>
#include <unordered_set>
#include <vector>

using namespace std;
using namespace pycs::gc;

int sum_range_29(int n)
{
    int total = 0;
    auto __range_start_i = 0;
    auto __range_stop_i = n;
    auto __range_step_i = 1;
    if (__range_step_i == 0) throw std::runtime_error("range() arg 3 must not be zero");
    for (auto i = __range_start_i; (__range_step_i > 0) ? (i < __range_stop_i) : (i > __range_stop_i); i += __range_step_i)
    {
        total = (total + i);
    }
    return total;
}

int main()
{
    py_print(sum_range_29(5));
    return 0;
}

File: test/cs/case25_for_range.cs
using System.Collections.Generic;
using System.IO;
using System;

public static class Program
{
    public static int sum_range_29(int n)
    {
        int total = 0;
        var __pytra_range_start_1 = 0;
        var __pytra_range_stop_2 = n;
        var __pytra_range_step_3 = 1;
        if (__pytra_range_step_3 == 0) throw new Exception("range() arg 3 must not be zero");
        for (var i = __pytra_range_start_1; (__pytra_range_step_3 > 0) ? (i < __pytra_range_stop_2) : (i > __pytra_range_stop_2); i += __pytra_range_step_3)
        {
            total = (total + i);
        }
        return total;
    }

    public static void Main(string[] args)
    {
        Pytra.CsModule.py_runtime.print(sum_range_29(5));
    }
}
